compiler/src/Language/Bond/Syntax/Internal.hs

```diff
@@ -51,3 +52,13 @@ metaField Field {..} = Any $ isMetaName fieldType
 isBaseField :: String -> Maybe Type -> Bool
 isBaseField name = getAny . optional (foldMapFields (Any.(name==).fieldName))
 
+-- If a Declaration is a Struct then return the Fields of its Parent
```

Comment:
There is a helper function `foldMapFields` you can use here:

    baseFields Struct{..} = foldMapFields return <$> structBase

The type of above is  `Maybe [Field]`. It will return `Nothing` if there is no base and list of fields, possibly empty, the whole base *hierarchy* (which is what I think you need). I think that `Maybe [Field]` is better than `[Field]` for the codegen (code for class with base that has no fields may look different than code for class with no base). But if you wanted to return `[]` in all cases then you could do:

    baseFields Struct{..} = concat $ foldMapFields return <$> structBase

Reply:
I found foldMapFields but I could not figure out the syntax to use it.  My Haskell is not good.  I've changed to using it.